src/core/public/chrome/ui/header/collapsible_nav.tsx

```diff
@@ -159,21 +183,22 @@ export function CollapsibleNav({
         onToggle={isCategoryOpen => setIsCategoryOpen('recentlyViewed', isCategoryOpen, storage)}
         data-test-subj="collapsibleNavGroup-recentlyViewed"
       >
-        {recentNavLinks.length > 0 ? (
+        {recentlyAccessed.length > 0 ? (
           <EuiListGroup
             aria-label={i18n.translate('core.ui.recentlyViewedAriaLabel', {
               defaultMessage: 'Recently viewed links',
             })}
-            // TODO #64541
-            // Can remove icon from recent links completely
-            listItems={recentNavLinks.map(({ iconType, onClick = () => {}, ...link }) => ({
-              'data-test-subj': 'collapsibleNavAppLink--recent',
-              onClick: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
-                onIsOpenUpdate(false);
-                onClick(e);
-              },
-              ...link,
-            }))}
+            listItems={recentlyAccessed.map(link => {
+              // TODO #64541
+              // Can remove icon from recent links completely
+              const { iconType, ...hydratedLink } = createRecentNavLink(link, navLinks, basePath);
```

Comment:
NIT: adding a `omitIcon = false` param to `createRecentNavLink` seems a little better than spread exclusion?